libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs

```diff
@@ -261,28 +292,101 @@ fn visit_scalar_field_attributes<'ast>(
                 }
              }
 
+             let model_db_name = model_data.mapped_name.unwrap_or(&ast_model.name.name);
+             let field_db_name = scalar_field_data.mapped_name.unwrap_or(&ast_field.name.name);
+
+             let db_name : Option<Cow<'ast, str>> = if ctx.db.preview_features.contains(NamedConstraints) {
+
+                 let generated_name = ConstraintNames::index_name(model_db_name, &[field_db_name], IndexType::Unique, ctx.db.active_connector());
+                 let db_name = get_map_argument(args, ctx, generated_name);
+                 validate_db_name(&ast_model, args, ctx, &db_name, "@unique");
+
+                 db_name
+
+             } else {
+                 let generated_name = if ctx.is_sql_server() {
+                     format!("{}_{}_unique", model_db_name, field_db_name)
+                 } else {
+                     format!("{}.{}_unique", model_db_name, field_db_name)
+                 };
+
+                 Some(Cow::from(generated_name))
+             };
+
+
             model_data.indexes.push((args.attribute(), IndexData {
                 is_unique: true,
                 fields: vec![field_id],
                 source_field: Some(field_id),
                 name: None,
-                db_name: None,
+                db_name,
             }))
          });
     });
 }
 
+fn primary_key_constraint_name<'ast>(
+    ast_model: &'ast Model,
+    model_data: &mut ModelData<'ast>,
+    args: &mut Arguments<'ast>,
+    ctx: &mut Context<'ast>,
+    attribute: &'ast str,
+) -> Option<Cow<'ast, str>> {
+    let generated_name = ConstraintNames::primary_key_name(
+        model_data.mapped_name.unwrap_or(&ast_model.name.name),
+        ctx.db.active_connector(),
+    );
+
+    let db_name = match args.optional_arg("map").map(|name| name.as_str()) {
+        Some(Ok("")) => error_on_empty_string_cow(args, ctx, "map"),
+        Some(Ok(name)) => Some(name.into()),
+        Some(Err(err)) => push_error_cow(ctx, err),
+        None if ctx.db.active_connector().supports_named_primary_keys() => Some(generated_name.into()),
+        None => None,
+    };
+
+    validate_db_name(&ast_model, args, ctx, &db_name, attribute);
+
+    if db_name.is_some() && !ctx.db.active_connector().supports_named_primary_keys() {
+        ctx.push_error(DatamodelError::new_model_validation_error(
+            "You defined a database name for the primary key on the model. This is not supported by the provider.",
+            &ast_model.name.name,
+            ast_model.span,
+        ));
+    }
+    db_name
+}
+
+fn validate_db_name<'ast>(
+    ast_model: &'ast Model,
+    args: &mut Arguments<'ast>,
+    ctx: &mut Context<'ast>,
```

Comment:
nitpick: by convention, I always pass the context argument last. It's unspoken so maybe we could coordinate on this in the engines meeting, but the general approach I think works well is to order arguments from "most specific" (what the function is actually about) to "less specific" (context, helpers, ...)

Reply:
I'm fine with standardizing on that. I adjusted it.